func/demo/filedescriptor.cpp

```diff
@@ -0,0 +1,198 @@
+#include "faasm/faasm.h"
+
+#include <cerrno>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <fcntl.h>
+#include <string>
+
+#include <unistd.h>
+
```

Comment:
I should have added a function like this when I wrote all the filesystem syscalls, but I did it in a bit of a rush (and it's regularly come back to bite me). It's meant to test the low-level handling of file descriptors, mainly `dup`, `seek` and `write`